platform/platform-logging/src/main/java/org/codice/ddf/logging/platform/LoggingServiceBean.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.logging.platform;
+
+import java.lang.management.ManagementFactory;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock.ReadLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock.WriteLock;
+
+import javax.management.InstanceAlreadyExistsException;
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanRegistrationException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.NotCompliantMBeanException;
+import javax.management.ObjectName;
+
+import org.ops4j.pax.logging.spi.PaxAppender;
+import org.ops4j.pax.logging.spi.PaxLoggingEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.EvictingQueue;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+/**
+ * A custom {@link org.ops4j.pax.logging.spi.PaxAppender} which receives 
+ * {@link org.ops4j.pax.logging.spi.PaxLoggingEvent}s and is the Jolokia 
+ * endpoint for the Logging UI in the Admin Console.
+ */
+public class LoggingServiceBean implements PaxAppender, LoggingServiceBeanMBean {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingServiceBean.class);
+
+    private static final String MBEAN_NAME = LoggingServiceBean.class.getName()
+            + ":service=logging-service";
+
+    private static final String BUNDLE_NAME = "bundle.name";
+
+    private static final String BUNDLE_VERSION = "bundle.version";
+
+    private static final int MAX_LOG_EVENTS_LIMIT = 5000;
+
+    private static final int DEFAULT_LOG_EVENTS_LIMIT = 500;
+
+    private final WriteLock writeLock = new ReentrantReadWriteLock().writeLock();
+
+    private final ReadLock readLock = new ReentrantReadWriteLock().readLock();
+
+    private EvictingQueue<LogEvent> logEvents;
+
+    private int maxLogEvents = DEFAULT_LOG_EVENTS_LIMIT;
+
+    private ObjectName objectName;
+
+    private MBeanServer mBeanServer;
+
+    public LoggingServiceBean() {
+        try {
+            logEvents = EvictingQueue.create(DEFAULT_LOG_EVENTS_LIMIT);
```

Comment:
Should we create the queue in the `init()` method instead and _only_ if it's not already been created by the `setMaxLogEvents`?